net/Socket.hpp

```diff
@@ -161,19 +161,13 @@ class Socket
     {
         LOG_TRC("Socket dtor");
 
-        // explicit closeFD called, or initial createSocket failure
-        if (_fd < 0)
-            return;
-
-        // Doesn't block on sockets; no error handling needed.
-#if !MOBILEAPP
-        ::close(_fd);
-        LOG_DBG("Closed socket " << toStringImpl());
-#else
-        fakeSocketClose(_fd);
-#endif
+        closeFD(); // In case we haven't closed yet.
     }
 
+    /// Returns true iff we have a valid FD and haven't called shutdown(2).
+    /// Note: this is needed because shutting down and closing are independent.
+    bool isOpen() const { return _fd >= 0 && !_isShutdown; }
+
     /// Returns true if this socket FD has been shutdown, but not necessarily closed.
     bool isShutdown() const { return _isShutdown; }
```

Comment:
Is it this still used.

Might want to remove it explicitly to remove mistakes.

Reply:
Yes. It's needed and used.